riocli/device/delete.py

```diff
@@ -26,26 +35,119 @@
+@click.option('--workers', '-w',
+              help="number of parallel workers while running delete devices "
+                   "command. defaults to 10.", type=int, default=10)
+@click.argument('device-name-or-regex', type=str, default="")
 @with_spinner(text='Deleting device...')
-def delete_device(device_name: str, device_guid: str, force: bool, spinner=None):
+def delete_device(
+        force: bool,
+        delete_all: bool,
+        workers: int,
+        device_name_or_regex: str,
+        spinner: Yaspin = None,
+) -> None:
     """
-    Deletes a device
+    Deletes one more devices
     """
+    client = new_client()
+    if not (device_name_or_regex or delete_all):
+        spinner.text = "Nothing to delete"
+        spinner.green.ok(Symbols.SUCCESS)
+        return
+
+    try:
+        devices = fetch_devices(
+            client, device_name_or_regex, delete_all)
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to delete device(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+    if not devices:
+        spinner.text = "No devices to delete"
+        spinner.ok(Symbols.SUCCESS)
+        return
+
+    headers = ['Name', 'Device ID', 'Status']
+    data = [[d.name, d.uuid, d.status] for d in devices]
+
     with spinner.hidden():
-        if not force:
-            click.confirm(
-                'Deleting device {} ({})'.format(
-                    device_name, device_guid), abort=True)
+        tabulate_data(data, headers)
+
+    spinner.write('')
+
+    if not force:
+        with spinner.hidden():
+            click.confirm('Do you want to delete above device(s)?',
+                          default=True, abort=True)
+        spinner.write('')
 
     try:
-        client = new_client(with_project=True)
-        client.delete_device(device_id=device_guid)
-        spinner.text = click.style('Device deleted successfully', fg=Colors.GREEN)
-        spinner.green.ok(Symbols.SUCCESS)
+        result = Queue()
+        func = functools.partial(_delete_deivce, client, result)
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            executor.map(func, devices)
+
+        result = sorted(list(result.queue), key=lambda x: x[0])
+
+        data, fg, statuses = [], Colors.GREEN, []
+        for name, status in result:
+            fg = Colors.GREEN if status else Colors.RED
+            icon = Symbols.SUCCESS if status else Symbols.ERROR
+            statuses.append(status)
+            data.append([
+                click.style(name, fg),
+                click.style(icon, fg)
+            ])
+
+        with spinner.hidden():
+            tabulate_data(data, headers=['Name', 'Status'])
+
+        icon = Symbols.SUCCESS if all(statuses) else Symbols.WARNING
+        fg = Colors.GREEN if all(statuses) else Colors.YELLOW
+        text = "successfully" if all(statuses) else "partially"
```

Comment:
This `partially` thing is not applicable to devices. Please re-write the logic for devices.